File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def to_s(m, s):
    return m * 60 + s

def to_m_s(s):
    m = s // 60
    s %= 60
    return m, s

def checking_opening(pos_s, op_start_s, op_end_s):
    if(op_start_s <= pos_s and pos_s <= op_end_s):
        return True
    return False
    
def next_cmd(video_len_s, pos_s):
    if(video_len_s - pos_s <= 10):
        return video_len_s
    return pos_s + 10

def prev_cmd(pos_s):
    if(pos_s < 10):
        return 0
    return pos_s - 10
    
def solution(video_len, pos, op_start, op_end, commands):
    video_len_s = to_s(int(video_len[0:2]), int(video_len[3:]))
    pos_s = to_s(int(pos[0:2]), int(pos[3:]))
    op_start_s = to_s(int(op_start[0:2]), int(op_start[3:]))
    op_end_s = to_s(int(op_end[0:2]), int(op_end[3:]))
    
    print(f'video_len_s: {video_len_s}')
    print(f'pos_s: {pos_s}')
    print(f'op_start_s: {op_start_s}')
    print(f'op_end_s: {op_end_s}\n')
    
    for c in commands:
        if(checking_opening(pos_s, op_start_s, op_end_s)):
            pos_s = op_end_s
        if(c == 'next'):
            pos_s = next_cmd(video_len_s, pos_s)
            if(checking_opening(pos_s, op_start_s, op_end_s)):
                pos_s = op_end_s
        else:   # c == 'prev'
            pos_s = prev_cmd(pos_s)
            if(checking_opening(pos_s, op_start_s, op_end_s)):
                pos_s = op_end_s
        print(f'command: {c}')
        print(f'pos_s: {pos_s}')
        
    m, s = map(str, to_m_s(pos_s))
    
    if(int(m) <= 9):
        m = '0' + m
    if(int(s) <= 9):
        s = '0' + s
    
    return m + ':' + s

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <vector>

using namespace std;

int to_s(int m, int s) { return 60 * m + s; }

string to_m_s(int sec) {
    int m = sec / 60;
    int s = sec % 60;
    
    string m_str = (m < 10 ? "0" : "") + to_string(m);
    string s_str = (s < 10 ? "0" : "") + to_string(s);
    
    return m_str + ":" + s_str;
}

bool checking_opening(int pos_s, int op_start_s, int op_end_s) {
    if(op_start_s <= pos_s && pos_s <= op_end_s) return true;
    return false;
}

int next_cmd(int video_len_s, int pos_s) {
    if(video_len_s - pos_s < 10) return 0;
    return pos_s + 10;
}

int prev_cmd(int pos_s) {
    if(pos_s < 10) return 0;
    return pos_s - 10;
}

string solution(string video_len, string pos, string op_start, string op_end, vector<string> commands) {
    string mmss[] = {video_len, pos, op_start, op_end};
    int s[4];
    int i = 0;
    
    for(string t : mmss) {
        string mm_str, ss_str;
        
        stringstream ss(t);
        getline(ss, mm_str, ':');
        getline(ss, ss_str, ':');
        
        s[i++] = to_s(stoi(mm_str), stoi(ss_str));
    }
    
    for(string cmd : commands) {
        if(checking_opening(s[1], s[2], s[3])) s[1] = s[3];
        if(cmd == "next") {
            s[1] = next_cmd(s[0], s[1]);
        }
        else if(cmd == "prev") {
            s[1] = prev_cmd(s[1]);
        }
        if(checking_opening(s[1], s[2], s[3])) s[1] = s[3];
    }
    
    return to_m_s(s[1]);
}